src/scmdata/processing.py

```diff
@@ -208,3 +217,227 @@ def calculate_exceedance_probabilities_over_time(
     out = _set_index_level_to(out, "unit", "dimensionless")
 
     return out
+
+
+def _set_peak_output_name(out, output_name, default_lead):
+    if output_name is not None:
+        out = _set_index_level_to(out, "variable", output_name)
+    else:
+        idx = out.index.names
+        out = out.reset_index()
+        out["variable"] = out["variable"].apply(
+            lambda x: "{} {}".format(default_lead, x)
+        )
+        out = out.set_index(idx)[0]
+
+    return out
+
+
+def calculate_peak(scmrun, output_name=None):
+    """
+    Calculate peak i.e. maximum of each timeseries
+
+    Parameters
+    ----------
+    scmrun : :class:`scmdata.ScmRun`
+        Ensemble of which to calculate the exceedance probability over time
+
+    output_name : str
+        If supplied, the value to put in the "variable" columns of the output
+        series. If not supplied, "Peak {variable}" will be used.
+
+    Returns
+    -------
+    :class:`pd.Series`
+        Peak of each timeseries
+    """
+    out = scmrun.timeseries().max(axis=1)
+    out = _set_peak_output_name(out, output_name, "Peak")
+
+    return out
+
+
+def calculate_peak_time(scmrun, output_name=None, return_year=True):
+    """
+    Calculate peak time i.e. the time at which each timeseries reaches its maximum
+
+    Parameters
+    ----------
+    scmrun : :class:`scmdata.ScmRun`
+        Ensemble of which to calculate the exceedance probability over time
+
+    output_name : str
+        If supplied, the value to put in the "variable" columns of the output
+        series. If not supplied, "Peak {variable}" will be used.
+
+    return_year : bool
+        If ``True``, return the year instead of the datetime
+
+    Returns
+    -------
+    :class:`pd.Series`
+        Peak of each timeseries
+    """
+    out = scmrun.timeseries().idxmax(axis=1)
+    if return_year:
+        out = out.apply(lambda x: x.year)
+
+    out = _set_peak_output_name(
+        out, output_name, "Year of peak" if return_year else "Time of peak"
+    )
+
+    return out
+
+
+def _calculate_quantile_groupby(base, index, quantile):
+    return base.groupby(index).quantile(quantile)
+
+
+def calculate_summary_stats(
+    scmrun,
+    index,
+    exceedance_probabilities_thresholds=[1.5, 2.0, 2.5],
+    exceedance_probabilities_variable="Surface Air Temperature Change",
+    exceedance_probabilities_naming_base=None,
+    peak_quantiles=[0.05, 0.17, 0.5, 0.83, 0.95],
+    peak_variable="Surface Air Temperature Change",
+    peak_naming_base=None,
+    peak_time_naming_base=None,
+    peak_return_year=True,
+    progress=False,
```

Comment:
Missing some documentation. Not sure if that is resolved in next PR or not

Reply:
oops good pickup